include/plstring.h

```diff
@@ -0,0 +1,251 @@
+#ifndef POKEPLATINUM_PLSTRING_H
+#define POKEPLATINUM_PLSTRING_H
+
+#include "struct_decls/struct_plstring_decl.h"
+
+/**
+ * Control codes for how to pad numeric strings with leading zeroes
+ * filling their allocation space.
+ * 
+ * Ex: `1000000` being assigned to a buffer of size 20 may be:
+ * - Left-justified:                      "1000000             "
+ * - Right-justified, padded with spaces: "             1000000"
+ * - Right-justified, padded with zeroes: "00000000000001000000"
+ */
+typedef enum {
+    NUMBER_PADDING_MODE_NONE,       ///< Left-justified strings; no padding
+    NUMBER_PADDING_MODE_SPACES,     ///< Right-justified strings, pad with spaces
+    NUMBER_PADDING_MODE_ZEROES      ///< Right-justified strings, pad with zeroes
+} NUMBER_PADDING_MODE;
+
+/**
+ * Control codes for which character-width should be used when translating
+ * numeric values into byte-data.
+ */
+typedef enum {
+    CHARACTER_WIDTH_FULL,           ///< Use full-width characters
+    CHARACTER_WIDTH_HALF            ///< Use half-width characters
+} CHARACTER_WIDTH;
```

Comment:
`typedef enum` is not allowed. Enum types should be PascalCase. Use `enum MyEnum` when referencing the type in source code.